doc/guides/test/sanitycheck.rst

```diff
@@ -548,15 +439,81 @@ Beside being able to run tests in QEMU and other simulated environments,
 sanitycheck supports running most of the tests on real devices and produces
 reports for each run with detailed FAIL/PASS results.
 
-To use this feature, run sanitycheck with the following new options::
+
+Executing tests on a single device
+===================================
+
+To use this feature on a single connected device, run sanitycheck with
+the following new options::
 
 	scripts/sanitycheck --device-testing --device-serial /dev/ttyACM0 -p \
 	frdm_k64f  -T tests/kernel
 
 The ``--device-serial`` option denotes the serial device the board is connected to.
 This needs to be accessible by the user running sanitycheck. You can run this on
-only one board at a time, specified using the
-``--platform`` option.
+only one board at a time, specified using the ``--platform`` option.
+
+
+Executing tests on multiple devices
+===================================
+
+To build and execute tests on multiple devices connected to the host PC a
+hardware map needs to be created with all connected devices and their
+details such as the serial device and their IDs if available. To produce the
+hardware map, please run the following command::
+
+    ./scripts/sanitycheck --generate-hardware-map map.yml
+
+The generated hardware map file (map.yml) will have the list of connected
+devices, for example::
+
+  - available: true
+    id: OSHW000032254e4500128002ab98002784d1000097969900
+    platform: unknown
+    product: DAPLink CMSIS-DAP
+    runner: pyocd
+    serial: /dev/cu.usbmodem146114202
+  - available: true
+    id: 000683759358
+    platform: unknown
+    product: J-Link
+    runner: unknown
+    serial: /dev/cu.usbmodem0006837593581
+
+
+Any options marked as 'unknown' need to be changed and set with the correct
+values, in the above example both the platform names and the runners need to be
+replaced with the correct values corresponding to the connected hardware. In
+this example we are using a reel_board and a nrf52840_pca10056::
+
+  - available: true
+    id: OSHW000032254e4500128002ab98002784d1000097969900
+    platform: reel_board
+    product: DAPLink CMSIS-DAP
+    runner: pyocd
+    serial: /dev/cu.usbmodem146114202
+  - available: true
+    id: 000683759358
+    platform: nrf52840_pca10056
+    product: J-Link
+    runner: nrfjprog
+    serial: /dev/cu.usbmodem0006837593581
+
+
+With the hardware map ready, you can run any tests by pointing to the map
+file::
+
+  ./scripts/sanitycheck --device-testing --hardware-map map.yml -T samples/hello_world/
+
+The above command will result in sanitycheck building tests for the platforms
+defined in the hardware map and subsequently flashing and running the tests
+on those platforms.
+
+.. note::
+
+  Currently only boards with support for both pyocd and nrfjprog are supported
```

Comment:
Do the boards need to support **both pyocd and nrfjprog** ? I'm guessing it's either, and if so, change this to:
```
Currently only boards with support for either pyocd or nrfjprog are supported ...
```